src/app/components/GridGallery/GridGallery.tsx

```diff
@@ -189,16 +123,62 @@ const GridGallery: React.FC = () => {
     document.body.classList.add('no-scroll');
     setOpenModal(index);
   }
+
+  const handlerCheckBox = (id: number) => {
+    if(id === selectedCategory){
+      setSelectedCategory(null);
+   }
+   else {
+    setSelectedCategory(id);
+   }
+  }
+
   return (
     <>
-      <div className="image-grid-container" id="gallery" ref={gridRef}>
-        {images.map((src, index) => (
-          <div className="grid-item" key={index}>
-            <img onClick={() => handleOpenModal(index)} src={src.images} alt={`Image ${index + 1}`} />
-          </div>
-        ))}
+      <div className='gird-filter-container'>
+        <div className='categories-checkboxes'>
+          {categories.map(item => {
+            return (
+              <div className='category-check-box' key={item.name}>
+                <p className='checkbox-label'>{item.name}</p>
+                <span className="checkbox-custom"></span>
+
+                <input 
+                  className='checkbox'
+                  type='checkbox'
+                  checked={selectedCategory === item.id}
+                  onChange={() => handlerCheckBox(item.id)}
+                  />
+              </div>
+            )
+          })}
+        </div>
+        <div className="image-grid-container" id="gallery" ref={gridRef}>
+          {nfts.length < 1 && 
+            <div className='no-results'> No results! </div>
+          }
+          {nfts.map((src, index) => {
+            return (
+              <div className="grid-item" key={index}>
+                <img onClick={() => handleOpenModal(index)} src={src?._embedded["wp:featuredmedia"][0]?.source_url} alt={src?._embedded["wp:featuredmedia"][0].alt_text} />
+              </div>
+            )
+          })}
+        </div>
       </div>
-      {openModal >= 0 && <Modal totalItems={images.length} activeIndex={openModal} handleNext={handleNext} handlePrev={handlePrev} item={images[openModal]} onClose={handleOnClose} />}
+      <button className='load-more' disabled={selectedCategory ? true : false} onClick={handleLoadMore}>
+        Load more
+      </button>
```

Comment:
Logic is fine but I prefer always to write descriptiive code so this can also go to separate function